Assets/Scripts/CrumblingFloorEvent.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CrumblingFloorEvent : MonoBehaviour
+{
+    public float crumbleDelay;
+
+    public AudioClip crumbling;
+
+    public AudioClip crumblingWarning;
+
+    private float timerCountDown = 1;
+
+    bool isColliding = false;
+
+    bool hasCrumblingAudioPlayed = false;
+
+
+    private void Update()
+    {
+        if (isColliding == true)
+        {
+            timerCountDown -= Time.deltaTime;
+            if (timerCountDown < 0)
+            {
+                timerCountDown = 0;
+            }
+        }
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
```

Comment:
use [OnCollision](https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnCollisionEnter2D.html) stuff instead of ontrigger. then you can get rid of the unnecessary child with the trigger.